src/ZEO/asyncio/client.py

```diff
@@ -199,9 +205,12 @@ def finish_connect(self, protocol_version):
 
     exception_type_type = type(Exception)
     def message_received(self, data):
-        msgid, async, name, args = decode(data)
+        msgid, async, name, args = self.decode(data)
         if name == '.reply':
             future = self.futures.pop(msgid)
+            if isinstance(future, tuple):
+                future = self.futures.pop(future)
```

Comment:
That's a good question. It's been so long, I don't remember for sure.

This is definitely wanted by the byte server so if not in this PR, then the next one.  Arguably, It makes the protocol cleaner as well.  Until fairly recently, msgid was always an integer. Then, when adding multiple simultaneous loads and prefetch, I added the hack of letting msgid be an (oid,tid) tuple. This hack worked because of the dynamic nature of Python. Rust, of course, is statically typed and it's a lot more convenient for msgid to always be an int. 

IIRC it *was* also helpful for msgpack, because by default, "arrays" are decoded as lists, which was messing up msgid lookup. I think I later change the default decoding for arrays to tuple, which would have made this a non-issue.